### ISSUE TITLE
Managing multiple versions of java

### ISSUE THREAD

**Issue by gramtech** [reedia/dotfiles#12]
After looking into managing java versions, there seems to be two options that make the most sense (to me)
See https://stackoverflow.com/questions/26252591/mac-os-x-and-multiple-java-versions

Have a look and let me know what you think

#### 1) ENV Vars for each with a default (quite clean)
<pre><code>
export JAVA_10_HOME=$(/usr/libexec/java_home -v10)
export JAVA_8_HOME=$(/usr/libexec/java_home -v1.8)
export JAVA_9_HOME=$(/usr/libexec/java_home -v9)

alias java10='export JAVA_HOME=$JAVA_10_HOME'
alias java8='export JAVA_HOME=$JAVA_8_HOME'
alias java9='export JAVA_HOME=$JAVA_9_HOME'

#default java
export JAVA_HOME=$JAVA_10_HOME
</code></pre>
Can easily change using the alias'd comands

#### 2) jenv (installed via brew)

This is a fairly comprehensive tool for managing java versions on your machine

Setup jenv (added to .bash_profile)
<pre><code>
if which jenv > /dev/null; then 
    eval "$(jenv init -)"
    export PATH="$HOME/.jenv/shims:$PATH"
fi
</code></pre>
Then get list of installed java versions
<pre><code>
/usr/libexec/java_home -V
</code></pre>
 Add each one to jenv
<pre><code>
jenv add /path/from/above/command
</code></pre>
you can then set then check and set the java versions Globally, locally or per folder etc

**Comment by sebastienrousseau** [comment]
@gramtech Sounds like a good solution, the main question to me is why would we need to have multiple java versions on a mac? I am already struggling to maintain multiple python or ruby environments... But anyway happy for you to bring it along. Saw a similar gist at https://gist.github.com/ntamvl/5f4dbaa8f68e6897b99682a395a44c2e

**Comment by gramtech** [comment]
You make a very good point, generally one would not need multiple versions of Java installed.
I have removed v8 and v9 from the brew install files and just doing a test to make sure that all selected brew packages are happy with the latest java (currentlly v10)

If anyone needs additional older version they can be installed manually as required.

Ill check this in soon and merge into master once testing is done.
